xarray/core/nanops.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from . import dtypes
+from .pycompat import dask_array_type
+from . duck_array_ops import (count, isnull, fillna, where_method,
+                              _dask_or_eager_func)
+from . import nputils
+
+try:
+    import dask.array as dask_array
+except ImportError:
+    dask_array = None
+
+
+def _replace_nan(a, val):
+    """
+    replace nan in a by val, and returns the replaced array and the nan
+    position
+    """
+    mask = isnull(a)
+    return where_method(val, mask, a), mask
+
+
+def _maybe_null_out(result, axis, mask, min_count=1):
+    """
+    xarray version of pandas.core.nanops._maybe_null_out
+    """
+    if hasattr(axis, '__len__'):  # if tuple or list
+        raise ValueError('min_count is not available for reduction '
+                         'with more than one dimensions.')
+
+    if axis is not None and getattr(result, 'ndim', False):
+        null_mask = (mask.shape[axis] - mask.sum(axis) - min_count) < 0
+        if np.any(null_mask):
+            dtype, fill_value = dtypes.maybe_promote(result.dtype)
+            result = result.astype(dtype)
+            result[null_mask] = fill_value
+
+    elif getattr(result, 'dtype', None) not in dtypes.NAT_TYPES:
+        null_mask = mask.size - mask.sum()
+        if null_mask < min_count:
+            result = np.nan
+
+    return result
+
+
+def _nan_argminmax_object(func, fill_value, value, axis=None, **kwargs):
+    """ In house nanargmin, nanargmax for object arrays. Always return integer
+    type """
+    valid_count = count(value, axis=axis)
+    value = fillna(value, fill_value)
+    data = getattr(np, func)(value, axis=axis, **kwargs)
```

Comment:
This function is doing eager evaluation by calling the numpy function -- can we call the dask function instead on dask arrays?

Alternatively, we could make this function error on dask arrays? I just don't want to silently compute them.